python/ray/serve/tests/unit/test_metrics_utils.py

```diff
@@ -1,16 +1,54 @@
+import asyncio
 import sys
 import time
+from typing import Optional
 from unittest.mock import patch
 
 import pytest
 
-from ray._private.test_utils import wait_for_condition
+from ray._private.test_utils import async_wait_for_condition
+from ray._private.utils import get_or_create_event_loop
 from ray.serve._private.metrics_utils import InMemoryMetricsStore, MetricsPusher
-from ray.serve._private.test_utils import MockTimer
+from ray.serve._private.utils import get_random_string
+
+
+class MockAsyncTimer:
+    def __init__(self, start_time: Optional[float] = 0):
+        self.async_sleeps = list()
+        self.reset(start_time=start_time)
+
+    def reset(self, start_time: 0):
+        self._curr = start_time
+
+    def time(self) -> float:
+        return self._curr
+
+    async def sleep(self, amt: float):
```

Comment:
Yes because multiple `asyncio.sleep(n)` executes concurrently, so we can't just do `self._curr += amt`. However I have simplified this compared to before, please take another look!